includes/class-amp-cli.php

```diff
@@ -0,0 +1,613 @@
+	 *
+	 * By default, this script only crawls URLs that support AMP,
+	 * where the user has not opted-out of AMP for the URL.
+	 * For example, by un-checking 'Posts' in 'AMP Settings' > 'Supported Templates'.
+	 * Or un-checking 'Enable AMP' in the post's editor.
+	 *
+	 * @var int
+	 */
+	public static $force_crawl_urls = false;
+
+	/**
+	 * A whitelist of conditionals to use for validation.
+	 *
+	 * Usually, this script will validate all of the templates that don't have AMP disabled.
+	 * But this allows validating based on only these conditionals.
+	 * This is set if the WP-CLI command has an --include argument.
+	 *
+	 * @var array
+	 */
+	public static $include_conditionals = array();
+
+	/**
+	 * The maximum number of URLs to validate for each type.
+	 *
+	 * Templates are each a separate type, like those for is_category() and is_tag().
+	 * Also, each post type is a separate type.
+	 * This value is overridden if the WP-CLI command has an --n argument, like --n=10.
+	 *
+	 * @var int
+	 */
+	public static $limit_type_validate_count;
+
+	/**
+	 * The validation counts by type, like template or post type.
+	 *
+	 * @var array[][] {
+	 *     Validity by type.
+	 *
+	 *     @type int $valid The number of valid URLs for this type.
+	 *     @type int $total The total number of URLs for this type, valid or invalid.
+	 * }
+	 */
+	public static $validity_by_type = array();
+
+	/**
+	 * Crawl the entire site to get AMP validation results.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--limit=<templates>]
+	 * : The maximum number of URLs to validate for each template and content type.
+	 * ----
+	 * default: 100
+	 * ----
+	 *
+	 * [--include=<templates>]
+	 * : Only validates a URL if one of the conditionals is true.
+	 *
+	 * [--force]
+	 * : Force validation of URLs even if their associated templates or object types do not have AMP enabled.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     wp amp validate-site --include=is_author,is_tag
+	 *
+	 * @subcommand validate-site
+	 * @param array $args       Positional args.
+	 * @param array $assoc_args Associative args.
+	 * @throws Exception If an error happens.
+	 */
+	public function validate_site( $args, $assoc_args ) {
+		unset( $args );
+		self::$include_conditionals      = array();
+		self::$force_crawl_urls          = false;
+		self::$limit_type_validate_count = (int) $assoc_args[ self::LIMIT_URLS_ARGUMENT ];
+
+		/*
+		 * Handle the argument and flag passed to the command: --include and --force.
+		 * If the self::INCLUDE_ARGUMENT is present, force crawling or URLs.
+		 * The WP-CLI command should indicate which templates are crawled, not the /wp-admin options.
+		 */
+		if ( ! empty( $assoc_args[ self::INCLUDE_ARGUMENT ] ) ) {
+			self::$include_conditionals = explode( ',', $assoc_args[ self::INCLUDE_ARGUMENT ] );
+			self::$force_crawl_urls     = true;
+		} elseif ( isset( $assoc_args[ self::FLAG_NAME_FORCE_VALIDATION ] ) ) {
+			self::$force_crawl_urls = true;
+		}
+
+		if ( ! current_theme_supports( 'amp' ) ) {
```

Comment:
This conditional will capture the Template Mode being Classic, though maybe you had in mind a more narrow check, like:
```php
'disabled' === AMP_Options_Manager::get_option( 'theme_support' )
```